.gitlab/llnl-ci-conda.yml

```diff
@@ -0,0 +1,117 @@
+stages:
+  - env-setup
+  - test
+  - cleanup
+
+.final_steps:
+  script: &finalize
+    - chgrp -fR exaworks $SDK_WORK_DIR/conda || true
+    - chmod -fR 02770 $SDK_WORK_DIR/conda    || true
+
+conda-env-setup:
+  stage: env-setup
+  parallel:
+    matrix:
+      - HOST : [ "quartz", "ruby"]
+        RUNNER_TYPE: "shell"
+        # - HOST : [ "lassen" ]
+        #   ENVIRONMENT: [ "BATCH" ]
+        #   RUNNER_TYPE: "batch"
+  tags:
+    - $HOST
+    - $RUNNER_TYPE
+  script:
+    - mkdir -p ${SDK_WORK_DIR}/conda/
+    - cd ${SDK_WORK_DIR}/conda/
+    - eval "$(conda shell.posix hook)"
+    - conda create --prefix ./${HOST}-exaworks-sdk-venv python=3.7 -y
+    - conda env list
+    - conda activate ./${HOST}-exaworks-sdk-venv
+    - conda env list
+    - ls
+    - pwd 
+    - conda config --add channels conda-forge
+    - conda config --add channels lightsource2-tag 
+    - conda install swift-t
+    - conda install parsl
+    - conda install radical.pilot
+    - pip install pytest
+    - pip install pytest-tap
+    - *finalize
+
+conda-sdk-clone:
+  stage: env-setup
+  needs: [ "conda-env-setup" ]
+  tags:
+    - quartz
+    - shell
+  script:
+    - cd ${SDK_WORK_DIR}/conda/
+    - git clone git@github.com:ExaWorks/SDK.git
+    - *finalize
+
+conda-tests:
+  stage: test
+  needs: [ "conda-sdk-clone" ]
+  parallel:
+    matrix:
+      - HOST : [ "quartz", "ruby"]
+        RUNNER_TYPE: "shell"
+        # - HOST : [ "lassen" ]
+        #   ENVIRONMENT: [ "BATCH" ]
+        #   RUNNER_TYPE: "batch"
+  tags:
+    - $HOST
+    - $RUNNER_TYPE
+  script:
+    - cd ${SDK_WORK_DIR}/conda/
+    - pwd 
+    - ls
+    - eval "$(conda shell.posix hook)"
+    - echo $(pwd)/${HOST}-exaworks-sdk-venv
+    - echo $CONDA_DEFAULT_ENV
+    - conda activate ./${HOST}-exaworks-sdk-venv
+    - echo $CONDA_DEFAULT_ENV
+    - conda info --envs
+    - conda list
+    - cd SDK
+    - export RADICAL_PILOT_DBURL=${MONGODB_CONNECTION_STRING}?tlsAllowInvalidCertificates=true
+    - export run_id=$CI_PIPELINE_ID
+    - export branch=$CI_COMMIT_BRANCH
+    - export url=$TESTING_HOST
+    - export test="Conda"
+    - export location="llnl-$HOST"
+    - export contact="arambula2@llnl.gov"
+    - export imnumber="LLNL-MI-834241"
+    - python3 ci/tests/test.py -s
+    - for TEST in parsl rp parsl-flux; do
+        python3 ci/tests/test.py -n $TEST -c "bash ci/tests/${TEST}/test.sh";
+      done
+    - python3 ci/tests/test.py -e
+
+.conda-env-cleanup: &conda-env-cleanup
+  stage: cleanup
+  needs: [ "conda-tests" ]
+  tags:
+    - quartz
+    - shell
+  script:
+    - cd $SDK_WORK_DIR
+    - rm -rf conda
```

Comment:
don't think we need this.. maybe have something similar what we have for spack? ->
* job-env-setup
  * if env exists then exit this job
  * otherwise create environment
* job-build (or install)
  * update packages (e.g., `conda env update -n <env_name> -f <environment_file_name>`)

Reply:
thus we don't need remove and recreate envs every time, rather just update packages

Reply:
I don't know of a way in which `conda` and `virtualenv` can create a environment if it doesn't exist, or use an existing one if it does, at least not with a one-line solution. Do you want me to add a bit of bash script to the `script` steps that checks if a environment exists, otherwise create one?

Reply:
something like this (that is what we have for spack as well)
```
pip-env-setup:
  stage: setup
  ...
  script:
    - test -d $PIP_WORK_DIR/$VENV_ENV_NAME && exit 0
    - python3 -m venv $PIP_WORK_DIR/$VENV_ENV_NAME
    - source $PIP_WORK_DIR/$VENV_ENV_NAME/bin/activate
    - pip install -U pip setuptools wheel
    - pip cache purge

pip-build:
  stage: setup
  needs:
    - job: pip-env-setup
  ...
  script:
    - export SRC_DIR=$(pwd)
    - source $PIP_WORK_DIR/$VENV_ENV_NAME/bin/activate
    - pip install --no-cache-dir -r .gitlab/venv-requirements.txt
```
and same for `conda`

Reply:
with such approach we also can split stage `setup` into two (thus to have four stages total), since all configs will follow it: setup (for spack it will include three jobs: spack-get, spack-env-setup, spack-update), build

Reply:
Done, split it into two different jobs

Should I do it also for conda?